OpenRA.Game/Traits/Target.cs

```diff
@@ -56,11 +56,11 @@ public enum TargetType : byte { Invalid, Actor, Terrain, FrozenActor }
 			generation = 0;
 		}
 
-		Target(Actor a)
+		Target(Actor a, int? generation = null)
```

Comment:
This feels a bit weird IMO, what if we instead have `Target(Actor a, int generation)` and a `Target(Actor a) : this(a, a.Generation) { }`?

Reply:
Sure, done.